_posts/2019-06-06-NeuralODEs.markdown

```diff
@@ -0,0 +1,388 @@
+     </p>
+     </details>
+
+<br />
+
+# 5 The adjoint method (and auto-diff)
+  **Motivation**: The adjoint method is a numerical method for efficiently computing the gradient of a function in numerical optimization problems. Understanding this method is essential to understand how to train ‘continuous depth’ nets. We also review the basics of Automatic Differentiation, which will help us understand the efficiency of the algorithm proposed in the NeuralODE paper. 
+
+  **Topics**:
+
+  1. Adjoint method
+  2. Auto-diff
+
+  **Required Reading**:
+
+  1. Section 8.7 from [Computational Science and Engineering](http://math.mit.edu/~gs/cse/) (CSE)
+  2. Sections 2,3 from [Automatic Differentiation in Machine Learning: a Survey](http://www.jmlr.org/papers/volume18/17-468/17-468.pdf)
+
+  **Notes**: Here is a [link](/assets/nodes_notes/week5.pdf) to our notes for the lesson. 
+  
+  **Optional Reading**:
+
+  1. [Prof. Steven G. Johnson's notes on adjoint method](http://math.mit.edu/~stevenj/notes.html)
+  
+  **Questions**:
+
+  1. Exercises 1,2,3 from Section 8.7 of CSE
+     <details><summary>Solution</summary>
+     <p>
+     Exercise 1:
+     This follows immediatly by noticing that the number of multiply-add operations of multiplying a \(N\times M\) matrix with a \(M\times P\) matrix is given by \(O(NMP)\). 
+     </p>
+     <p>
+     Exercise 2:
+     We simply need to apply the chain rule. Since \(\frac{\partial C}{\partial S} = 2S\) and \(\frac{dT}{dS} = \frac{\partial T}{\partial S} + \frac{\partial T}{\partial C}\frac{\partial C}{\partial S}\), we get \(\frac{d T}{d S} = 1 -2S\).  
+     </p>
+     <p>
+     Exercise 3:
+     This follows by Exercise 1 by seeing \(u^T\) and \(w^T\) as \(1\times N\) matrices and \(v\) as a \(N\times 1\) matrix. 
+     </p>
+     </details>
+
+  2. Consider the problem of optimizing a real-valued function \(g\) over the solution of the ODE \(y' = Ay\), \(y(0) = y_0\) at time \(T>0\): \(min_{y0, A} g(y(T))\). What is the solution of the adjoint equation?
+     <details><summary>Solution</summary>
+     <p>
+     See notes below.
+     </p>
+     </details>
+
+  3. How do you get eq. (14) in Section 8.7 of CSE?
+     <details><summary>Solution</summary>
+     <p>
+     See notes below.
+     </p>
+     </details>
+
+<br />
+
+# 6 Neural ODEs
+  **Motivation**: Let’s read the paper!
+
+  **Topics**:
+
+  1. Normalising Flows
+  2. End-to-end implementations with neural nets
+
+  **Required Reading**:
+
+  1. [Neural Ordinary Differential Equations](https://arxiv.org/abs/1806.07366)
+  2. [A blog post on NeuralODEs](https://rkevingibson.github.io/blog/neural-networks-as-ordinary-differential-equations/)
+  
+  **Notes**: Here is a [link](/assets/nodes_notes/week6.pdf) to our notes for the lesson. 
```

Comment:
This is such a good opportunity to discuss what was discussed in class.

Reply:
what are you suggesting here?

Reply:
Week 6 is rather sparse on content because it's mostly about the paper. It's a good opportunity to talk about what the group found interesting and focused on during the session.

Reply:
I added a brief description after the section title, and a brief description of what's in the notes.

Reply:
Much better, thanks!
- "Here is a summary of what’s going on to help with your understanding:"
- "allows to avoid" --> "allows us to avoid"
- "allowing to easily" --> "allowing us to easily"

Reply:
Further below:
- "rispective adjoint" --> "respective adjoint"
- "then discussed continuous normalising flows and understood the computational advantages offered by Neural ODEs" --> get rid of "understood".

Reply:
fixed